runelite-client/src/main/java/net/runelite/client/plugins/blastmine/BlastMineOreCountOverlay.java

```diff
@@ -49,39 +50,43 @@
 
 	private final Client client;
 	private final BlastMinePluginConfig config;
-	private final ImagePanelComponent imagePanelComponent = new ImagePanelComponent();
+	private final ItemManager itemManager;
+	private final PanelComponent panelComponent = new PanelComponent();
 
 	@Inject
-	private ItemManager itemManager;
-
-	@Inject
-	BlastMineOreCountOverlay(Client client, BlastMinePluginConfig config)
+	BlastMineOreCountOverlay(Client client, BlastMinePluginConfig config, ItemManager itemManager)
 	{
 		setPosition(OverlayPosition.TOP_LEFT);
 		this.client = client;
 		this.config = config;
+		this.itemManager = itemManager;
+		panelComponent.setOrientation(PanelComponent.Orientation.HORIZONTAL);
 	}
 
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
 		Widget blastMineWidget = client.getWidget(WidgetInfo.BLAST_MINE);
 
-		imagePanelComponent.getImages().clear();
+		panelComponent.getChildren().clear();
 
 		if (blastMineWidget != null)
 		{
 			blastMineWidget.setHidden(true);
 			if (config.showOreOverlay())
 			{
-				imagePanelComponent.getImages().add(getImage(COAL, client.getSetting(Varbits.BLAST_MINE_COAL)));
-				imagePanelComponent.getImages().add(getImage(GOLD_ORE, client.getSetting(Varbits.BLAST_MINE_GOLD)));
-				imagePanelComponent.getImages().add(getImage(MITHRIL_ORE, client.getSetting(Varbits.BLAST_MINE_MITHRIL)));
-				imagePanelComponent.getImages().add(getImage(ADAMANTITE_ORE, client.getSetting(Varbits.BLAST_MINE_ADAMANTITE)));
-				imagePanelComponent.getImages().add(getImage(RUNITE_ORE, client.getSetting(Varbits.BLAST_MINE_RUNITE)));
+				panelComponent.getChildren().add(new ImageComponent(getImage(COAL, client.getVar(Varbits.BLAST_MINE_COAL))));
+				panelComponent.getChildren().add(new ImageComponent(getImage(GOLD_ORE, client.getVar(Varbits.BLAST_MINE_GOLD))));
+				panelComponent.getChildren().add(new ImageComponent(getImage(MITHRIL_ORE, client.getVar(Varbits.BLAST_MINE_MITHRIL))));
+				panelComponent.getChildren().add(new ImageComponent(getImage(ADAMANTITE_ORE, client.getVar(Varbits.BLAST_MINE_ADAMANTITE))));
+				panelComponent.getChildren().add(new ImageComponent(getImage(RUNITE_ORE, client.getVar(Varbits.BLAST_MINE_RUNITE))));
 			}
 		}
-		return imagePanelComponent.render(graphics);
+		if (!panelComponent.getChildren().isEmpty())
```

Comment:
No need to check for empty, panel component should handle that. Also add empty line above.